packages/hardhat/test/ChainLance.ts

```diff
@@ -13,15 +13,240 @@ describe("ChainLance", function () {
+    const txAcceptBid = await chainLance.connect(employer).acceptBid(projectId, bidId, { value: projectPrice });
+    const receiptAcceptBid = await txAcceptBid.wait();
+    expect(receiptAcceptBid?.status).is.equal(1, "bid creation failed");
+
+    ///////////////////////////////////////////////////////////////////
+    // Submit work
+    ///////////////////////////////////////////////////////////////////
+    const txSubmitWork = await chainLance.connect(worker).submitWork(projectId);
+    const receiptSubmitWork = await txSubmitWork.wait();
+    expect(receiptSubmitWork?.status).is.equal(1, "Work submition failed");
+    ///////////////////////////////////////////////////////////////////
+    // Accept Work
+    ///////////////////////////////////////////////////////////////////
+    const txAcceptWork = await chainLance.connect(employer).acceptWork(projectId);
+    const receiptAcceptWork = await txAcceptWork.wait();
+    expect(receiptAcceptWork?.status).is.equal(1, "Work acception failed");
+    if (receiptAcceptWork) {
+      expect(await ethers.provider.getBalance(worker.address)).is.greaterThanOrEqual(
+        BigInt(projectPrice) - receiptAcceptWork.gasPrice * receiptAcceptWork.gasUsed,
+        "Worker did not recieve payment",
+      );
+    }
+
+    // Worker has 0 balance edge case accounted with gtThanOrEq
+  });
+
+  it("Should allow cancelling/rejecting during the work on the project", async function () {
+    const [employer, worker] = await ethers.getSigners();
+
+    const projectId = "id2";
+    const projectPrice = 200000;
+    const timeSpanToCancel = 100;
+    const txCreateProject = await chainLance.connect(employer).createProject(projectId, projectPrice, timeSpanToCancel);
+    const receiptCreateProject = await txCreateProject.wait();
+
+    expect(receiptCreateProject).is.not.null;
+    expect(receiptCreateProject?.status).is.equal(1, "project creation failed");
+
+    const bidId = "bid2";
+    await chainLance.connect(worker).bidProject(projectId, bidId, projectPrice, timeSpanToCancel - 1); // timespan on project is for bidding?
+
+    await chainLance.connect(employer).acceptBid(projectId, bidId, { value: projectPrice });
+
+    await chainLance.connect(worker).submitWork(projectId);
+
+    ///////////////////////////////////////////////////////////////////
+    // Reject work submition
+    ///////////////////////////////////////////////////////////////////
+    const txRejectWork = await chainLance.connect(employer).rejectWork(projectId);
+    const receiptRejectWork = await txRejectWork.wait();
+    expect(receiptRejectWork?.status).is.equal(1, "work rejection failed");
+
+    ///////////////////////////////////////////////////////////////////
+    // Cancel work
+    ///////////////////////////////////////////////////////////////////
+
+    await ethers.provider.send("evm_increaseTime", [timeSpanToCancel + 1]); // Changing block timestamp
+    await ethers.provider.send("evm_mine", []); // Mine block
```

Comment:
We don't need this 2 lines here. We are waiting for next block in  `txRejectWork.wait();` If you need wait for more than 1 block,  you can pass value in this function,  for example `txRejectWork.wait(2);` But I don't understand why we need to wait for block at all,  any number of operations on project cancel should work inside 1 block.

Reply:
txRejectWork.wait(timeSpanToCancel + 1); results in error: "Error: Timeout of 40000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves"

Reply:
Let's set `timeSpanToCancel` to 1 for cancel work test